hs_core/hydroshare/resource.py

```diff
@@ -662,6 +681,21 @@ def add_resource_files(pk, *files, **kwargs):
         # no file has been added, make sure data/contents directory exists if no file is added
         utils.create_empty_contents_directory(resource)
     else:
+        if resource.resource_type == "CompositeResource" and auto_aggregate:
+            from hs_file_types.utils import set_logical_file_type
+            # check folders for aggregations
+            for fol in new_folders:
+                folder = resource.short_id + "/data/contents/" + fol
+                agg_type = resource.get_folder_aggregation_type_to_set(folder)
+                if agg_type is not None and agg_type is not '':
+                    agg_type = agg_type.replace('LogicalFile', '')
+                    set_logical_file_type(res=resource, user=None, file_id=None,
+                                          hs_file_type=agg_type, folder_path=fol,
+                                          fail_feedback=False)
+            # check files for aggregation
+            for res_file in ret:
+                set_logical_file_type(res=resource, user=None, file_id=res_file.pk,
```

Comment:
May be better to check if the file is not already part of a logical file before creating an aggregation from it.
```if not res_file.has_logical_file:```

Reply:
done